com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs

```diff
@@ -39,28 +39,7 @@ internal class Styles
         Material[]      m_MaterialLayers = new Material[kMaxLayerCount];
         AssetImporter   m_MaterialImporter;
 
-        int numLayer
-        {
-            get { return (int)layerCount.floatValue; }
-            set
-            {
-                layerCount.floatValue = (float)value;
-                UpdateEditorExpended(value);
-            }
-        }
-
-        void UpdateEditorExpended(int layerNumber)
-        {
-            if (layerNumber == 4)
-            {
-                materialEditor.SetExpandedAreas((uint)ExpandableBit.ShowLayer3, true);
-            }
-            if (layerNumber >= 3)
-            {
-                materialEditor.SetExpandedAreas((uint)ExpandableBit.ShowLayer2, true);
-            }
-            materialEditor.SetExpandedAreas((uint)ExpandableBit.ShowLayer1, true);
-        }
```

Comment:
Be careful with this. You require extend test with HDRP/LayeredLit and HDRP/LayeredLitTesselation
This ShowLayerX are not for expended but for reallly showing the LayerX

Reply:
This code was only called in the setter, and the setter was never being called, besides that the code for a layer number 4 will expand to `ShowLayer2` as the condition is wrong.

Reply:
Truth. Code should have changed